dist/js/header/headerController.js

```diff
@@ -1,24 +1,100 @@
 export default class HeaderController {
-    constructor(headerMenuContainer, headerHeading) {
-        this.headerMenuContainer = headerMenuContainer;
-        this.headerHeading = headerHeading;
+  constructor(headerMenuContainer, headerHeading) {
+    this.headerMenuContainer = headerMenuContainer;
+    this.headerHeading = headerHeading;
+    this.previousYCoordinate = 0;
+    this.scrollEventExecutionFlag = false;
+    this.headerMenuContainerSize = 0;
+    this.menuShowFlag = false;
+  }
+
+  init() {
+    this.getHeaderElementSize();
+    this.pageScrollListener();
+  }
+
+  pageScrollListener() {
+    document.addEventListener('scroll', () => {
+      if (!this.scrollEventExecutionFlag) {
+        requestAnimationFrame(this.scrollEvent.bind(this));
+        this.scrollEventExecutionFlag = true;
+      }
+    });
+  }
+
+  scrollEvent() {
+    const scrollTopSafeSpace = 50; // 50px scroll top to show sticky menu
+    const docElement = document.documentElement;
```

Comment:
docElement should be initialized once on instance init

Reply:
fixed